src/controllers/dailyReportCommand.ts

```diff
@@ -0,0 +1,240 @@
+import { Logger } from 'structured-log';
+import * as discord from 'discord.js';
+import { IActiveCollabAPI } from './activecollab-api';
+import { ICommandController } from './command';
+import * as ProjectTasks from '../models/projectTasks';
+import * as userController from './userController';
+import { IMappingController } from './mapping';
+
+export async function dailyReport(
+    projects: string[],
+    eventColor: any,
+    activeCollabApi: IActiveCollabAPI,
+    logger: Logger,
+    writeToCsv: (
+        columNames: string[],
+        rows: string[][],
+        filename: string,
+        logger: Logger
+    ) => Promise<void>
+): Promise<Array<discord.RichEmbed>> {
+
+    const messages: discord.RichEmbed[] = [];
+
+    for (const projectID of projects) {
+        let taskData: ProjectTasks.TasksData;
+        try {
+            taskData = await activeCollabApi
+                .getAssignmentTasksByProject(projectID);
+        } catch (e) {
+            const errorMsg = `Error getting tasks for project[${projectID}]`;
+            logger.error(errorMsg + `: ${e}`);
+            return [
```

Comment:
This will terminate the retrieval of task data for any subsequent projectID, maybe push to the array instead?